wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java

```diff
@@ -1010,21 +1010,37 @@ private void cleanup() {
 
         executor.shutdown();
 
+        boolean interrupted = false;
+        final ExecutorService destroyMachinesExecutor = Executors.newCachedThreadPool(
```

Comment:
The suggestion is to use `FixedThreadPool` executor service instead, as you can control the count of threads(btw it caches them as well). What if you have 100 machines running ?  Also, i think that it is necessary to increase termination timeout, or at least make it configurable along with count of threads.